website/docs/r/host_virtual_switch.html.markdown

```diff
@@ -0,0 +1,173 @@
+
+## Example Usages
+
+**Create a virtual switch with one active and one standby NIC:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics  = ["vmnic0"]
+  standby_nics = ["vmnic1"]
+}
+```
+
+**Create a virtual switch with extra networking policy options:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics    = ["vmnic0"]
+  standby_nics   = ["vmnic1"]
+  teaming_policy = "failover_explicit"
+
+  allow_promiscuous      = false
+  allow_forged_transmits = false
+  allow_mac_changes      = false
+
+  shaping_enabled           = true
+  shaping_average_bandwidth = 50000000
+  shaping_peak_bandwidth    = 100000000
+  shaping_burst_size        = 1000000000
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (String, required, forces new resource) The name of the virtual switch.
+* `host_system_id` - (String, required, forces new resource) The managed object
+  ID of the host to set the virtual switch up on. 
+* `mtu` - (Integer, optional) The maximum transmission unit (MTU) for the virtual
+  switch. Default: `1500`.
+* `number_of_ports` - (Integer, optional) The number of ports to create with
+  this virtual switch. Default: `128`.
+
+~> **NOTE:** Changing the port count requires a reboot of the host. Terraform
+will not restart the host for you.
+
+### Bridge Options
+
+The following arguments are related to how the virtual switch binds to physical
+NICs:
+
+* `network_adapters` - (Array of strings, required) The network interfaces to
+  bind to the bridge.
+* `beacon_interval` - (Integer, optional) The interval, in seconds, that a NIC
+  beacon packet is sent out. This can be used with
+  [`check_beacon`](#check_beacon) to offer link failure capability beyond link
+  status only. Default: `1`.
+* `link_discovery_operation` - (String, optional) Whether to `advertise` or
+  `listen` for link discovery traffic. Default: `listen`.
+* `link_discovery_protocol` - (String, optional) The discovery protocol type.
+  Valid types are `cpd` and `lldp`. Default: `cdp`.
+
+### Policy Options
+
+The following options relate to how network traffic is handled on this virtual
+switch. It also controls the NIC failover order. This subset of options is
+shared with the [`vsphere_host_port_group`][host-port-group] resource, in which
```

Comment:
`[`vsphere_host_port_group`][host-port-group]` - does the second one want brackets?

Reply:
Naw, links with references are `[text][ref]` - kind of like the numbered form (`[text][1]`).